rust-runtime/smithy-json/src/escape.rs

```diff
@@ -5,35 +5,38 @@
 
 use std::borrow::Cow;
 
-const ESCAPES: &[char] = &['"', '\\', '\u{08}', '\u{0C}', '\n', '\r', '\t'];
-
 /// Escapes a string for embedding in a JSON string value.
 pub fn escape_string(value: &str) -> Cow<str> {
-    if !value.contains(ESCAPES) {
-        return Cow::Borrowed(value);
+    let bytes = value.as_bytes();
+    for (index, byte) in bytes.iter().enumerate() {
+        match byte {
+            0..=0x1F | b'"' | b'\\' => {
+                return Cow::Owned(escape_string_inner(&bytes[0..index], &bytes[index..]))
+            }
+            _ => {}
+        }
     }
+    Cow::Borrowed(value)
+}
 
-    let mut escaped = String::new();
-    let (mut last, end) = (0, value.len());
-    for (index, chr) in value
-        .char_indices()
-        .filter(|(_index, chr)| ESCAPES.contains(chr))
-    {
-        escaped.push_str(&value[last..index]);
-        escaped.push_str(match chr {
-            '"' => "\\\"",
-            '\\' => "\\\\",
-            '\u{08}' => "\\b",
-            '\u{0C}' => "\\f",
-            '\n' => "\\n",
-            '\r' => "\\r",
-            '\t' => "\\t",
-            _ => unreachable!(),
-        });
-        last = index + 1;
+fn escape_string_inner(start: &[u8], rest: &[u8]) -> String {
+    let mut escaped = start.to_vec();
+    for byte in rest {
+        match byte {
+            b'"' => escaped.extend("\\\"".bytes()),
```

Comment:
did you use `bytes()` here instead of `b` to make it clear that these were valid strings before being turned into bytes?

Reply:
No, I just didn't think to use `b`.